src/server/process_request.c

```diff
@@ -617,13 +617,24 @@ process_request(int sfds)
 	if (server.sv_attr[(int)SRV_ATR_acl_host_enable].at_val.at_long) {
 		/* acl enabled, check it; always allow myself	*/
 		
-		if ((acl_check(&server.sv_attr[(int)SRV_ATR_acl_hosts],
-			request->rq_host, ACL_Host) == 0) &&
-			(strcasecmp(server_host, request->rq_host) != 0)) {
-				req_reject(PBSE_BADHOST, 0, request);
-				close_client(sfds);
-				return;
+		struct pbsnode *isanode = NULL;
+		if ((server.sv_attr[SRV_ATR_acl_host_moms_enable].at_flags & ATR_VFLAG_SET) &&
+			(server.sv_attr[(int)SRV_ATR_acl_host_moms_enable].at_val.at_long == 1)) {
+			isanode = find_nodebyaddr(get_connectaddr(sfds));
+
+			if (isanode->nd_state & INUSE_DELETED)
```

Comment:
since find_nodebyaddr() can return NULL, please check for isanode to be a non NULL address before dereferencing it .